src/cluster_slot_stats.c

```diff
@@ -0,0 +1,173 @@
+            i++;
+        }
+    }
+    qsort(slot_stats, i, sizeof(slotStatEntry), (desc) ? slotStatEntryDescCmp : slotStatEntryAscCmp);
+}
+
+static void addReplySlotStat(client *c, int slot) {
+    addReplyLongLong(c, slot);
+    addReplyMapLen(c, 1);
+    addReplyBulkCString(c, "key-count");
+    addReplyLongLong(c, countKeysInSlot(slot));
+}
+
+static void addReplySlotStats(client *c, unsigned char *assigned_slots, int startslot, int endslot, int len) {
+    addReplyMapLen(c, len);
+
+    for (int slot = startslot; slot <= endslot; slot++) {
+        if (assigned_slots[slot]) addReplySlotStat(c, slot);
+    }
+}
+
+static void addReplySortedSlotStats(client *c, slotStatEntry slot_stats[], long limit) {
+    int num_slots_assigned = getMyShardSlotCount();
+    int len = min(limit, num_slots_assigned);
+    addReplyMapLen(c, len);
+
+    for (int i = 0; i < len; i++) {
+        addReplySlotStat(c, slot_stats[i].slot);
+    }
+}
+
+static void sortAndAddReplySlotStats(client *c, int order_by, long limit, int desc) {
+    slotStatEntry slot_stats[CLUSTER_SLOTS];
+    collectAndSortSlotStats(slot_stats, order_by, desc);
+    addReplySortedSlotStats(c, slot_stats, limit);
+}
+
+void clusterSlotStatsCommand(client *c) {
+    if (server.cluster_enabled == 0) {
+        addReplyError(c,"This instance has cluster support disabled");
+        return;
+    }
+
+    /* Parse additional arguments. */
+    if (c->argc == 5 && !strcasecmp(c->argv[2]->ptr,"slotsrange")) {
+        /* CLUSTER SLOT-STATS SLOTSRANGE start-slot end-slot */
+        int startslot, endslot;
+        if ((startslot = getSlotOrReply(c,c->argv[3])) == C_ERR ||
+            (endslot = getSlotOrReply(c,c->argv[4])) == C_ERR) {
+            return;
+        }
+        if (startslot > endslot) {
+            addReplyErrorFormat(c,"Start slot number %d is greater than end slot number %d", startslot, endslot);
+            return;
+        }
+        /* Initialize slot assignment array. */
+        unsigned char assigned_slots[CLUSTER_SLOTS]= {UNASSIGNED_SLOT};
+        int len = 0;
+        markSlotsAssignedToMyShard(assigned_slots, startslot, endslot, &len);
+        addReplySlotStats(c, assigned_slots, startslot, endslot, len);
+
+    } else if (c->argc >= 4 && !strcasecmp(c->argv[2]->ptr,"orderby")) {
+        /* CLUSTER SLOT-STATS ORDERBY metric [LIMIT limit] [ASC | DESC] */
+        int desc = 1, order_by = INVALID;
+        if (!strcasecmp(c->argv[3]->ptr, "key-count")) {
+            order_by = KEY_COUNT;
+        } else {
+            addReplyError(c, "Unrecognized sort metric for ORDER BY. The supported metrics are: key-count.");
+            return;
+        }
+        int i = 4; /* Next argument index, following ORDERBY */
+        int limit_counter = 0, asc_desc_counter = 0;
+        long limit;
+        while(i < c->argc) {
+            int moreargs = c->argc > i+1;
+            if (!strcasecmp(c->argv[i]->ptr,"limit") && moreargs) {
```

Comment:
```suggestion
            if (!strcasecmp(c->argv[i]->ptr, "limit") && moreargs) {
```
I think the formatter will complain about this, and some other code. So would be good to get the CI running by fixing the merge.